src/sortStudents.ts

```diff
@@ -1,16 +1,104 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'avgGrade'
+}
+
+function compareBools(a: boolean, b: boolean): number {
+  if (a === b) {
+    return 0;
+  }
+
+  if (a) {
+    return -1;
+  }
+
+  return 1;
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
+
+export function sortStudents(
+  students: Student[], sortBy: SortType, order: SortOrder,
+): Student[] {
+  const resultArray = [...students];
+
+  const average = (arr: number[]): number => {
+    return arr.reduce((p: number, c: number) => p + c, 0) / arr.length;
+  };
 
+  switch (sortBy) {
+    case 'name':
+      if (order === 'desc') {
+        resultArray.sort(
+          (a: Student, b: Student): number => b.name.localeCompare(a.name),
```

Comment:
these types are probably infered and you don't need to write them
```suggestion
          (a, b): number => b.name.localeCompare(a.name),
```